src/components/discovery/DiscoveryQueryBuilder.js

```diff
@@ -169,24 +171,29 @@ class DiscoveryQueryBuilder extends Component {
         });
 
         const addConditionsOnDataType = (buttonProps = {style: {float: "right"}}) => (
-            <Dropdown overlay={dataTypeMenu}
-                      disabled={this.props.dataTypesLoading || this.props.searchLoading}>
+            <Dropdown
+                overlay={dataTypeMenu}
+                disabled={this.props.dataTypesLoading || this.props.searchLoading || filteredDataTypes?.length === 0 }>
                 <Button {...buttonProps}> <Icon type="plus" /> Data Type <Icon type="down" /></Button>
             </Dropdown>
         );
 
         return <Card style={{marginBottom: "1.5em"}}>
             <DataTypeExplorationModal
-                dataTypes={this.props.dataTypes}
+                dataTypes={filteredDataTypes}
                 visible={this.state.schemasModalShown}
                 onCancel={this.handleHelpAndSchemasToggle}
             />
 
             <Typography.Title level={3} style={{marginBottom: "1.5rem"}}>
                 Advanced Search
                 {addConditionsOnDataType()}
-                <Button style={{float: "right", marginRight: "1em"}}
-                        onClick={this.handleHelpAndSchemasToggle}><Icon type="question-circle" /> Help</Button>
+                <Button
+                    style={{float: "right", marginRight: "1em"}}
+                    disabled={filteredDataTypes?.length === 0}
+                    onClick={this.handleHelpAndSchemasToggle}
+                    ><Icon type="question-circle" /> Help
```

Comment:
reformat this line (newline after the first `>`)

Reply:
Sure!